ddcz/views/legacy.py

```diff
@@ -0,0 +1,136 @@
+from django.apps import apps
+from ddcz.models.used.creations import Creation, CreativePage
+from django.http import HttpResponseRedirect
+from django.http.response import HttpResponse
+from django.shortcuts import get_object_or_404
+from django.urls.base import reverse
+from django.core.exceptions import ObjectDoesNotExist
+from django.views.decorators.http import require_http_methods
+
+# In the constant below: It is possible to vary the redirections.
+# It is preferable to do so via the constants rather than changing
+# the code if there is no real need for it, so the legacy router
+# remains automatized.
+
+CREATION = ["prispevky", "prispevky_komp"]
+CREATION_DETAIL = "prispevky_precti"
+
+# Forbidden dictionary keys from CREATION and CREATION_DETAIL
+LEGACY_PLAIN_ROUTER = {
+    "aktuality": "ddcz:news",
+    "novinky": "ddcz:newsfeed",
+    "uzivatele": "ddcz:users-list",
+    "registrace": "ddcz:sign-up",
+    "seznamka": "ddcz:dating",
+    "forum": "ddcz:phorum-list",
+    "linky": "ddcz:links-list",
+    "inzerce": "ddcz:market",
+    "credits": "ddcz:web-authors-and-editors",
+    "faq": "ddcz:website-manual",
+    "oao": "ddcz:faq",
+    "cojetodrd": "ddcz:about-drd",
+}
+
+LEGACY_LEGACY_CREATION_ROUTER = {
+    "hrbitov": "hrbitov",
+    "clanky": "clanky",
+    "expanze": "expanze",
+    "kouzelnikzvl": "kouzelnikzvl",
+    "alchymistazvl": "alchymistazvl",
+    "valecnik": "valecnik",
+    "hranicar": "hranicar",
+    "zlodej": "zlodej",
+    "novapovolani": "novapovolani",
+    "noverasy": "noverasy",
+}
+
+
+LEGACY_CREATION_ROUTER = [
+    "bestiar",
+    "dobrodruzstvi",
+    "predmety",
+    "kouzla",
+    "alchpredmety",
+    "hranicarkouzla",
+    "dovednosti",
+    "galerie",
+    "fotogalerie",
+]
+
+
+@require_http_methods(["GET"])
+def legacy_router(request):
+
+    get = dict(request.GET)
+    section = handle_get_key(get, "rub")
+    subsection = handle_get_key(get, "co")
+    id = handle_get_key(get, "id")
+
+    # The LEGACY_PLAIN_ROUTER is redirecting basic pages.
+    # Typically no creative pages are present here.
+    if section in LEGACY_PLAIN_ROUTER.keys():
+        return HttpResponseRedirect(reverse(LEGACY_PLAIN_ROUTER[section]))
+
+    # Some of the creation have legacy url
+    # index.php?rub=prispevky(_komp)&co=page_slug
+    # Here are lists of such creative pages.
+    if section in CREATION and subsection in LEGACY_LEGACY_CREATION_ROUTER.keys():
+        return HttpResponseRedirect(
+            reverse(
+                "ddcz:creation-list",
+                kwargs={
+                    "creative_page_slug": LEGACY_LEGACY_CREATION_ROUTER[subsection]
+                },
+            )
+        )
+
+    # For index.php?rub=prispevky_jeden&subsection=page_slug&id=article_id
+    # we can find the article by Id. Problem would be, if the ID
+    # if articles can be changed, than this will lead to the new
```

Comment:
What do you mean by new article? 

But otherwise no, ID identifies an article by definition

Reply:
Then it's fine and I'll delete the comment. :smile_cat: